contracts/src/internal/HyperdriveCheckpoint.sol

```diff
@@ -238,28 +248,32 @@ abstract contract HyperdriveCheckpoint is
         (uint256 lpSharePrice, ) = _calculateLPSharePriceSafe(_vaultSharePrice);
         emit CreateCheckpoint(
             _checkpointTime,
+            checkpointVaultSharePrice,
             _vaultSharePrice,
             maturedShortsAmount,
             maturedLongsAmount,
             lpSharePrice
         );
 
-        return _vaultSharePrice;
+        return checkpointVaultSharePrice;
     }
 
     /// @dev Calculates the proceeds of the holders of a given position at
     ///      maturity.
     /// @param _bondAmount The bond amount of the position.
     /// @param _vaultSharePrice The current vault share price.
```

Comment:
These natspec params are out of order. i think it should be:

        @param  _bondAmount,
        @param  _openVaultSharePrice,
        @param  _closeVaultSharePrice,
        @param  _vaultSharePrice,    <--- this one is out of order in the code
        @param  _isLong